ui/api/endpoint.go

```diff
@@ -224,6 +224,96 @@ func removeChannelEndpoint(svc ui.Service) endpoint.Endpoint {
 	}
 }
 
+func connectThingEndpoint(svc ui.Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		cr := request.(connectThingReq)
+
+		if err := cr.validate(); err != nil {
+			return nil, err
+		}
+
+		res, err := svc.Connect(ctx, cr.token, []string{cr.ChanID}, []string{cr.ThingID})
+		if err != nil {
+			return nil, err
+		}
+		return uiRes{
+			html: res,
+		}, err
+	}
+}
+
+func connectEndpoint(svc ui.Service) endpoint.Endpoint {
```

Comment:
What's the difference of `connectEndpoint` and other 2 functions?

Reply:
connectEndpoint is for listing all thing connections (button connections), connectChannelEndpoint is for listing all channel connections, and connectThingEndpoint is for connecting thing to channel and opposite.

Reply:
Then the naming is not really good. Should be `listConnections` or something, as we use for Users and Things - AFAIK we are using `list-` prefix, right?

Reply:
Accordingly, structs in `req` and `resp` should be renamed.

Reply:
Also, why are we using `svc.ViewConnections()` here - should it not be `svc.ViewConnections()` according to our standard, similar to [this](https://github.com/mainflux/mainflux/blob/master/things/api/things/http/endpoint.go#L173) (and same for `viewResourceReq`)